h2/src/main/org/h2/expression/function/Function.java

```diff
@@ -1534,6 +1540,42 @@ protected Value getValueWithArgs(Session session, Expression[] args) {
         return result;
     }
 
+    private Value regexpSubstr(Value inputString, Value regexpArg, Value positionArg, 
+            Value occurrenceArg, Value regexpModeArg, Value subexpressoinArg) {
+        String regexp = regexpArg.getString();
+
+        if (regexpArg == ValueNull.INSTANCE || positionArg == ValueNull.INSTANCE 
+            || occurrenceArg == ValueNull.INSTANCE || subexpressoinArg == ValueNull.INSTANCE) {
+            return ValueNull.INSTANCE;
+        }
+
+        int position = positionArg != null ? positionArg.getInt() - 1 : 0;
+        int requestedOccurrence = occurrenceArg != null ? occurrenceArg.getInt() : 1;
+        String regexpMode = regexpModeArg != null ? regexpModeArg.getString() : null;
+        int subexpression = subexpressoinArg != null && subexpressoinArg != ValueNull.INSTANCE ? subexpressoinArg.getInt() : 0;
+        int flags = makeRegexpFlags(regexpMode, false);
+        try {
+            Matcher m = Pattern.compile(regexp, flags).matcher(inputString.getString());
+
+            boolean found = m.find(position);
+            for(int occurrence = 1; occurrence < requestedOccurrence && found; occurrence++) {
+                found = m.find();
+            }
+
+            if (!found) {
+                return ValueNull.INSTANCE;
+            }
+            else {
+                return ValueVarchar.get(m.group(subexpression));
+            }
+        } catch (PatternSyntaxException e) {
+            throw DbException.get(ErrorCode.LIKE_ESCAPE_ERROR_1, e, regexp);
+        }
+        catch (IndexOutOfBoundsException e) {
+            return ValueNull.INSTANCE;
+        }
```

Comment:
Please, add test cases for both exceptions.

Reply:
33b1736